platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultProjectDependency.java

```diff
@@ -103,19 +102,32 @@ public ProjectDependency copy() {
     }
 
     @Override
+    @Deprecated
     public Set<File> resolve() {
         return resolve(true);
     }
 
     @Override
+    @Deprecated
+    @SuppressWarnings("deprecation")
     public Set<File> resolve(boolean transitive) {
-        CachingDependencyResolveContext context = new CachingDependencyResolveContext(taskDependencyFactory, transitive, Collections.emptyMap());
+
+        DeprecationLogger.deprecate("Directly resolving a project dependency's files")
```

Comment:
🤔 I think it would be better to include the name of the project dep that is being directly resolved here.  Yes, this might produce multiple similar deprecation warnings, but it will greatly benefit the maintainer to immediately see which project dep is responsible - this is first question anyone will have when they see this warning.

Reply:
Also for `getBuildDependencies()`.

Reply:
Added the project name to the message